.github/workflows/release.yml

```diff
@@ -0,0 +1,82 @@
+name: Create Release
+
+on:
+  push:
+    # Sequence of patterns matched against refs/tags
+    tags:
+      - '[0-9].[0-9]+.[0-9]+' # Push events to any matching semantic tag. For example, 1.10.1 or 2.0.0.
+        # For more details, see https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet
+        # and https://docs.npmjs.com/about-semantic-versioning
+
+jobs:
+  build:
+    runs-on: ${{matrix.os}}
+
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        node-version: [12.x, 14.x]
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Cache node modules
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-node-modules
+        with:
+          # npm cache files are stored in `~/.npm` on Linux/macOS
+          path: ~/.npm
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package.json') }}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ env.cache-name }}-
+            ${{ runner.os }}-build-
+            ${{ runner.os }}-
+
+      - name: Test
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+
+      - run: npm install
+      - run: cp env.example .env
+      - run: npm run test
+  Prettier-Check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v1
+      - run: npm install && npm run prettier-check
+  Lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v1
+      - run: npm install && npm run eslint
+  release:
+    needs: [build, Prettier-Check, Lint]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
```

Comment:
Should we do this on 23 above too?

Reply:
That's a good question. I will need to do some research to answer this

Reply:
@humphd so after looking up the docs the answer seems to be no. [`fetch-depth: 0`](https://github.com/actions/checkout/blob/main/README.md#checkout-v2) is needed for the `checkout` action to fetch all history for all branches and tags. In the `release` job we want that because we need to generate a changelog of everything that happened to the codebase since the last tag. On line 23 we want to fetch the latest version of code and run tests on it, no need to run tests on every version of code that existed since the last tag. 

P.S. Took a look at [this PR](https://github.com/actions/checkout/pull/258) and t[his issue](https://github.com/actions/checkout/issues/204) to get some context about the `fetch-depth`